otelx/otlp.go

```diff
@@ -81,9 +81,9 @@ func getTraceExporter(c *TracerConfig) (*otlptrace.Exporter, error) {
 
 	if c.Providers.OTLP.Protocol == "grpc" {
 		// Set up a connection to the OTLP server.
-		conn, err := grpc.DialContext(ctx, c.Providers.OTLP.ServerURL, grpc.WithTransportCredentials(insecure.NewCredentials()))
+		conn, err := grpc.NewClient(c.Providers.OTLP.ServerURL, grpc.WithTransportCredentials(insecure.NewCredentials()))
 		if err != nil {
-			errors.Errorf("failed to connect to OTLP gRPC endpoint: %s", err)
+			errors.Errorf("failed to connect to OTLP gRPC endpoint: %s", err) //nolint:errcheck,gosec
```

Comment:
Pretty sure I commented this on the last PR as well, we should not disable the lint here as it's valid. We should return the error here

```suggestion
			return nil, errors.Errorf("failed to connect to OTLP gRPC endpoint: %s", err)
```